src/Form/DataTransformer/ModelToIdTransformer.php

```diff
@@ -30,33 +32,51 @@ class ModelToIdTransformer implements DataTransformerInterface
 
     /**
      * @var string
+     *
+     * @phpstan-var class-string<T>
      */
     protected $className;
 
     /**
      * @param string $className
+     *
+     * @phpstan-param class-string<T> $className
      */
     public function __construct(ModelManagerInterface $modelManager, $className)
     {
         $this->modelManager = $modelManager;
         $this->className = $className;
     }
 
-    public function reverseTransform($newId)
+    /**
+     * @param mixed|null $value
+     *
+     * @return object|null
+     *
+     * @phpstan-return T|null
+     */
+    public function reverseTransform($value)
     {
-        if (empty($newId) && !\in_array($newId, ['0', 0], true)) {
+        if (empty($value) && !\in_array($value, ['0', 0], true)) {
             return null;
         }
 
-        return $this->modelManager->find($this->className, $newId);
+        return $this->modelManager->find($this->className, $value);
     }
 
-    public function transform($model)
+    /**
+     * @param object|null $value
+     *
+     * @return mixed|null
```

Comment:
I think that `getNormalizedIdentifier` return `string`.

```suggestion
     * @return string|null
```